Cargo.toml

```diff
@@ -26,8 +26,10 @@ volatile = "0.4.4"
 rustversion = "1.0.5"
 
 [features]
-default = ["nightly", "instructions"]
+default = ["nightly", "instructions", "amd_sev"]
```

Comment:
I'm not sure if we should enable this by default. Most users won't be running their code in an SEV VM, and there's likely a non-zero performance impact.

People have also been using this crate with AMD SEV-SNP and Intel TDX for a while, even without explicit support in this crate, and we don't want to accidentally break their code. In practice, on all currently available platforms, the C-bit/S-bit is always an address bit (I think this is even guaranteed on Intel TDX). People have been considering it as such and not a page table bit, so they're just setting the bit in `PhysAddr`/`PhysFrame` directly.

Reply:
You're right - it should not be enabled by default, I will fix.

The reason I'm making this patch is that I tried running an unmodified Hermit OS in an QEMU SEV VM, and it did not work, because of the c-bit.
In practice, I could retract the patch and do what you suggest with physical address bits in Hermit directly.

Reply:
Hermit OS will need to be modified anyway; SEV/SEV-ES/SEV-SNP/TDX all require kernel modifications.

Reply:
FYI: At Hermit we are also working on SEV support, but as it indeed requires numerous kernel modifications, this is not (yet) ready for publication. We also made our own quick-and-dirty changes to this crate - not too different from what is proposed here, but also not yet ready and generic. So a clean and upstream version of SEV support in this crate would be much appreciated :+1:

Reply:
That's great to hear. Out of curiosity, what generation are you targeting, SEV, SEV-ES, or SEV-SNP?

Reply:
> FYI: At Hermit we are also working on SEV support, but as it indeed requires numerous kernel modifications, this is not (yet) ready for publication. We also made our own quick-and-dirty changes to this crate - not too different from what is proposed here, but also not yet ready and generic. So a clean and upstream version of SEV support in this crate would be much appreciated 👍

Oh, is there any way we could get in touch? I should probably have started with that, but better late than never.

Reply:
> That's great to hear. Out of curiosity, what generation are you targeting, SEV, SEV-ES, or SEV-SNP?

Well, SEV without SNP doesn't make much sense anymore, right? :grin:

> Oh, is there any way we could get in touch? I should probably have started with that, but better late than never.

Sure, just drop us a mail :wink: